appservice/src/intTest/groovy/nu/yona/server/ActivityCommentTest.groovy

```diff
@@ -0,0 +1,552 @@
+		def richardMessagesRevisited = getActivityDetailMessages(richardResponseDetails, richard, expectedDataRichard5, 10).responseData._embedded."yona:messages"
+
+		then:
+		richardMessagesRevisited[0].nickname == "BD"
+		richardMessagesRevisited[1].nickname == "RQ (me)"
+		richardMessagesRevisited[2].nickname == "BD"
+		richardMessagesRevisited[3].nickname == "BDD"
+		richardMessagesRevisited[4].nickname == "RQ (me)"
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+		appService.deleteUser(bea)
+	}
+
+	def 'Comment on inactive day'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+
+		Goal budgetGoalNewsBuddyBob = richard.buddies[0].findActiveGoal(NEWS_ACT_CAT_URL)
+
+		def responseDayOverviewsBobAsBuddyAll = appService.getDayActivityOverviews(richard, richard.buddies[0])
+		assert responseDayOverviewsBobAsBuddyAll.status == 200
+
+		def responseDetailsBobAsBuddy = appService.getDayDetailsFromOverview(responseDayOverviewsBobAsBuddyAll, richard, budgetGoalNewsBuddyBob, 0, getCurrentShortDay(YonaServer.now))
+		assert responseDetailsBobAsBuddy.responseData._links."yona:addComment".href
+		assert responseDetailsBobAsBuddy.responseData._links."yona:messages".href
+
+		when:
+		def message = """{"message": "You're quiet!"}"""
+		def responseAddMessage = appService.yonaServer.createResourceWithPassword(responseDetailsBobAsBuddy.responseData._links."yona:addComment".href, message, richard.password)
+
+		then:
+		responseAddMessage.status == 200
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+	}
+
+	def 'Comment on inactive week'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+
+		Goal budgetGoalNewsBuddyBob = richard.buddies[0].findActiveGoal(NEWS_ACT_CAT_URL)
+
+		def responseWeekOverviewsBobAsBuddyAll = appService.getWeekActivityOverviews(richard, richard.buddies[0])
+		assert responseWeekOverviewsBobAsBuddyAll.status == 200
+
+		def responseDetailsBobAsBuddy = appService.getWeekDetailsFromOverview(responseWeekOverviewsBobAsBuddyAll, richard, budgetGoalNewsBuddyBob, 0)
+		assert responseDetailsBobAsBuddy.responseData._links."yona:addComment".href
+		assert responseDetailsBobAsBuddy.responseData._links."yona:messages".href
+
+		when:
+		def message = """{"message": "You're quiet!"}"""
+		def responseAddMessage = appService.yonaServer.createResourceWithPassword(responseDetailsBobAsBuddy.responseData._links."yona:addComment".href, message, richard.password)
+
+		then:
+		responseAddMessage.status == 200
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+	}
+
+	def 'Richard can delete a comment of Bob on which Richard already replied'()
```

Comment:
How about "Richard successfully deletes a comment..."? Otherwise, the test name becomes very long.